wandb/sdk/wandb_artifacts.py

```diff
@@ -466,13 +467,32 @@ def verify(self, root: Optional[str] = None):
             "Cannot call verify on an artifact before it has been logged or in offline mode"
         )
 
-    def save(self) -> None:
+    def save(self, **init_kwargs) -> None:  # type: ignore
         if self._logged_artifact:
             return self._logged_artifact.save()
-
-        raise ValueError(
-            "Cannot call save on an artifact before it has been logged or in offline mode"
-        )
+        else:
+            if wandb.run is None:
+                kwargs = {k: v for k, v in init_kwargs.items()}
+                if "job_type" not in init_kwargs:
+                    kwargs["job_type"] = "auto"
+                if "settings" not in init_kwargs:
+                    kwargs["settings"] = wandb.Settings(silent=True)
+                with wandb.init(**kwargs) as run:
+                    run.log_artifact(self)
+                    project_url = run._get_project_url()
+                termlog(
```

Comment:
I *think we can get the current version after calling `log_artifact` now.  If we're linking we may as well link directly to the version we've created.

Reply:
Yeah, we will have to cal a .wait(), but since the run is explicitly 'finished' here anyway, we have to wait for the upload - i'll add that